src/NuGet.Configuration/Utility/SettingsUtility.cs

```diff
@@ -130,16 +130,16 @@ public static string GetGlobalPackagesFolder(ISettings settings)
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            // Note that the value GlobalPackagesFolder must be a full path and not a relative path
-            // Read it like a string and explicitly not as a path
-            var path = settings.GetValue(ConfigSection, GlobalPackagesFolderKey, isPath: false);
+            var path = Environment.GetEnvironmentVariable(GlobalPackagesFolderEnvironmentKey);
             if (!string.IsNullOrEmpty(path))
             {
                 path = path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
```

Comment:
The replace/return should be done in one place, these can both use the same code path once it picks where to get the value from.